lib/mayaUsd/fileio/translators/translatorMaterial.cpp

```diff
@@ -425,6 +428,38 @@ bool UsdMayaTranslatorMaterial::AssignMaterial(
                 }
             }
         }
+
+        // We now have sufficient information to transform material SdfPaths to equivalent Maya
+        // object identifiers.
+        JsValue info;
+        if (UsdMayaMeshReadUtils::getGeomSubsetInfo(shapeObj, info) && info) {
+            JsObject meshRoundtrippingInfo = info.GetJsObject();
+            JsObject updatedInfo;
+
+            for (auto&& ssInfoIt : meshRoundtrippingInfo) {
+                JsObject subsetRoundtrippingInfo = ssInfoIt.second.GetJsObject();
+                auto     materialIt
+                    = subsetRoundtrippingInfo.find(UsdMayaGeomSubsetTokens->MaterialPathKey);
+                if (materialIt != subsetRoundtrippingInfo.cend()) {
+                    SdfPath materialPath(materialIt->second.GetString());
+                    MObject materialObj = context->GetMayaNode(materialPath, false);
+                    if (!materialObj.isNull()) {
+                        MFnDependencyNode depNodeFn(materialObj);
+                        // Names are not yet final at this stage of import. The only reliable
+                        // identifier is the UUID.
+                        subsetRoundtrippingInfo[UsdMayaGeomSubsetTokens->MaterialUuidKey]
+                            = JsValue(depNodeFn.uuid().asString().asChar());
+                    }
+                    subsetRoundtrippingInfo.erase(materialIt);
+                    updatedInfo[ssInfoIt.first] = subsetRoundtrippingInfo;
+                }
```

Comment:
We don't report not finding material in JSON or Maya? The user might want to know that some binding were lost? They might notice it visually, but they may not in a large scene.